src/target/riscv/riscv-013.c

```diff
@@ -3195,7 +3197,17 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
 					}
 					keep_alive();
 					dmi_status_t status = dmi_scan(target, NULL, &sbvalue[next_read_j],
-												   DMI_OP_READ, sbdata[j], 0, false);
+								       DMI_OP_READ, sbdata[j], 0, false);
+					if (j == 0 && info->bus_master_read_delay) {
```

Comment:
This part was harder for me to understand:
```suggestion
					/* By reading from sbdata0, we have just initiated another system bus read.
					 * Wait to let the read finish. */
					if (j == 0 && info->bus_master_read_delay) {
```

Reply:
Done. Thanks.

Yes, I have spike changes that exercise this new code. I'm working on cleaning them up and will probably submit them over there next week.